SCons/Executor.py

```diff
@@ -306,30 +306,30 @@ def get_all_children(self):
         over and over), so removing the duplicates once up front should
         save the Taskmaster a lot of work.
         """
-        result = SCons.Util.UniqueList([])
+        result = []
         for target in self.get_all_targets():
             result.extend(target.children())
-        return result
+        return SCons.Util.uniquer_hashables(result)
 
     def get_all_prerequisites(self):
         """Returns all unique (order-only) prerequisites for all batches
         of this Executor.
         """
-        result = SCons.Util.UniqueList([])
+        result = []
         for target in self.get_all_targets():
             if target.prerequisites is not None:
                 result.extend(target.prerequisites)
-        return result
+        return SCons.Util.uniquer_hashables(result)
```

Comment:
If possible I'd like to keep this as is. Right now this doesn't really change semantics, so it is hard to see how it could cause breakage, outside of a caller expecting unique-ification to be preserved if they mutate the result. But *someone* once thought these functions should retain order, so I'd prefer not to second guess that. I'd be happy to see such a change in a subsequent PR.